extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java

```diff
@@ -124,7 +120,9 @@ private static QuarkusSecurityIdentity augmentIdentity(SecurityIdentity security
     }
 
     private CompletionStage<SecurityIdentity> performCodeFlow(IdentityProviderManager identityProviderManager,
-            RoutingContext context) {
+            RoutingContext context, DefaultTenantConfigResolver resolver) {
+        TenantConfigContext configContext = resolver.resolve(context, true);
```

Comment:
It looks like resolve is now called lots of times per request. I think it might make sense to add some code to DefaultTenantResolver that stores the result in the routing context so subsequent invocations are effectively cached.

Reply:
It is only a visual impression :-). This PR only introduces one extra resolver call in `OidcAuthenticationMechanism` to figure which way to go. It is called a few times in `CodeAuthenticationMechanism` right on the master requesting a complete resolution. This PR optimizes it a bit, asking for `OidcTenantConfig` only when no `OAuth2Auth` is needed.
And @pedroigor has already updated `DefaultTenantConfigResolver` to support caching `TenantConfigContext` when needed. So it should be fine